src/nanoemoji/write_font.py

```diff
@@ -273,26 +274,38 @@ def _update_paint_glyph(paint):
         reuse_result = glyph_cache.try_reuse(path_in_font_space)
         if reuse_result is not None:
             # TODO: when is it more compact to use a new transforming glyph?
+            child_transform = Affine2D.identity()
             child_paint = paint.paint
-            if is_transform(child_paint) and is_gradient(child_paint.paint):
-                # We have a transformed gradient so we need to reverse the effect of the
+            if is_transform(child_paint):
+                child_transform = child_paint.gettransform()
+                child_paint = child_paint.paint
+
+            assert fixed_safe(*reuse_result.transform)
+            overflows = False
+            if is_gradient(child_paint):
```

Comment:
What if there is a gradient that isn't the direct child?

I don't think we actually *DO* this so maybe a TODO would suffice.

Reply:
thanks for noticing that. Indeed, we do not. The code only handles gradients that are direct child of PaintGlyph or child of child of PaintGlyph, via PaintTransform, which is the only subtree that we currently produce. I'll add the TODO